src/ServiceInsight/MessageList/MessageListView.xaml.cs

```diff
@@ -56,7 +56,11 @@ void OnBeforeLayoutRefresh(object sender, CancelRoutedEventArgs e)
 
         void SortData(ColumnBase column, ListSortDirection order)
         {
-            Model.RefreshMessages(column.Tag as string, order == ListSortDirection.Ascending);
+            var orderBy = column.Tag as string;
+            var ascending = order == ListSortDirection.Ascending;
+
+            Model.ChangeSorting(orderBy, ascending);
+            Model.RefreshMessages();
```

Comment:
Change sorting is a local op, does not go to the server. The API now looks something like:
```
void Search(Endpoint endpoint, string searchQuery = null) //starts a brand new search from page 1
void NavigateToPage(string link, int pageIndex) //Continues a search using link returned from previous request
void RefreshMessages() //Starts a brand new search using values already present in the view model
```